samples/bluetooth/central_gatt_write/src/gatt_write_common.c

```diff
@@ -12,10 +12,132 @@
 #include <zephyr/bluetooth/gatt.h>
 #include <zephyr/bluetooth/hci.h>
 
+/* Count down number of metrics intervals before performing a PHY update */
+#define PHY_UPDATE_COUNTDOWN 3U
+static uint32_t phy_update_countdown;
+static uint8_t phy_param_idx;
+
+void phy_update_iterate(struct bt_conn *conn)
+{
+	const struct bt_conn_le_phy_param phy_param[] = {
+		/* List of 1M Tx with Rx on other PHYs */
+		{
+			.options = BT_CONN_LE_PHY_OPT_NONE,
+			.pref_tx_phy = BT_GAP_LE_PHY_1M,
+			.pref_rx_phy = BT_GAP_LE_PHY_1M,
+		}, {
+			.options = BT_CONN_LE_PHY_OPT_NONE,
+			.pref_tx_phy = BT_GAP_LE_PHY_1M,
+			.pref_rx_phy = BT_GAP_LE_PHY_2M,
+		}, {
+			.options = BT_CONN_LE_PHY_OPT_NONE,
+			.pref_tx_phy = BT_GAP_LE_PHY_1M,
+			.pref_rx_phy = BT_GAP_LE_PHY_CODED,
+
+		/* List of 2M Tx with Rx on other PHYs */
+		}, {
+			.options = BT_CONN_LE_PHY_OPT_NONE,
```

Comment:
Kind of minor, but it's slightly awkward that the comment about subsequent array elements ends up being inside the block of a previous array element. In this case I'd have formatted it like:
```c
},

/* List of 2M Tx with Rx on other PHYs */
{
	.options = BT_CONN_LE_PHY_OPT_NONE,
	...
```

Reply:
done